### PR TITLE
[JSONRPC] add script_hash beside ScriptView

### PR DESCRIPTION
**Pull request by thefallentree**
Also , change arg_auth_key_prefix to. "auth_key_prefix"

### CODE REVIEW

File: json-rpc/src/tests/unit_tests.rs [hunk 1]
```diff
@@ -469,6 +486,15 @@ fn test_get_account_transaction_impl(
             let data: Option<TransactionView> = fetch_data(resp);
             let tx_view = data.expect("Transaction didn't exists!");
 
+            let expected_tx = mock_db
+                .all_txns
+                .iter()
+                .filter(|x| x.as_signed_user_txn().is_ok())
+                .map(|x| x.as_signed_user_txn().unwrap())
+                .find(|x| x.sender() == *acc && x.sequence_number() == seq)
```
Comment (christinacelee) [inline json-rpc/src/tests/unit_tests.rs]: nit, can ignore: I feel this can be more concise

File: json-rpc/src/views.rs [hunk 2]
```diff
@@ -179,21 +180,22 @@ impl TransactionView {
 pub enum ScriptView {
     #[serde(rename = "peer_to_peer_transaction")]
     PeerToPeer {
+        script_hash: String,
```
Comment (phoenix-antigravity) [inline json-rpc/src/views.rs]: if it's going to be common field for all enum values, do you think we should move it one level up to UserTransaction?

File: json-rpc/src/tests/unit_tests.rs (hunk 1 above)
Comment (thefallentree) [reply json-rpc/src/tests/unit_tests.rs]: fixed

File: json-rpc/src/tests/unit_tests.rs [hunk 3]
```diff
@@ -15,6 +15,7 @@ use futures::{channel::mpsc::channel, StreamExt};
 use hex;
 use libra_config::utils;
 use libra_crypto::ed25519::*;
```
[the remark this answers is not in the document]
Comment (thefallentree) [reply json-rpc/src/tests/unit_tests.rs]: done

File: json-rpc/src/views.rs (hunk 2 above)
Comment (thefallentree) [reply json-rpc/src/views.rs]: done

**Comment by thefallentree** [comment]
all fixed, PTAL

**Pull request closed without merging**